packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/create-page.js

```diff
@@ -25,7 +30,11 @@ module.exports = function (pageComponent) {
 
   if (matchingThemes.length == 1) {
     const [theme] = matchingThemes
-    const [, component] = pageComponent.split(path.join(theme.themeDir, `src`))
+
+    const pageComponentWithoutExtension = pathWithoutExtension(pageComponent)
+    const [, component] = pageComponentWithoutExtension.split(
+      path.join(theme.themeDir, `src`)
+    )
```

Comment:
Did this work in Windows before? It does now because both parts are normalized through `path.join`. I guess that's good but implicit and easy to break.